adam/src/main/kotlin/com/malinskiy/adam/extension/List.kt

```diff
@@ -16,4 +16,4 @@
 
 package com.malinskiy.adam.extension
 
-fun List<String>.bashEscape() = "'" + joinToString(" ") { it.replace("\'", "\\'") } + "'"
\ No newline at end of file
+fun List<String>.bashEscape() = "'" + joinToString(" ") { it.replace("'", "'\\''") } + "'"
```

Comment:
Just escaping the quote inside single quotes doesn't do what you want, you need to end the single quotes, insert it, and start single quotes again.